### PR TITLE
feat: store columns while saving view and restore columns on selecting view without select columns

### PR DESCRIPTION
**Pull request by ahmadshaheer**
### Summary
- store the selected columns in list view of traces while saving a view
- if user selects a view that doesn't have any selectColumns associated with it, restore the most recent select columns

#### Related Issues / PR's
close https://github.com/SigNoz/signoz/issues/4801
close https://github.com/SigNoz/signoz/issues/5882

#### Screenshots
Updated Loom Video
https://www.loom.com/share/b25e5b5ddd7440ff97980edcc453fbc4

#### E2E tests
https://github.com/SigNoz/signoz-e2e/pull/8

### CODE REVIEW

**Review by rahulkeswani101 — APPROVED** [review]

**Review by SagarRajput-7 — DISMISSED** [review]

**Review by vikrantgupta25 — APPROVED** [review]

**Review by YounixM — DISMISSED** [review]

File: frontend/src/container/ExplorerOptions/ExplorerOptions.tsx [hunk 1]
```diff
@@ -252,6 +256,38 @@ function ExplorerOptions({
...
+		aggregateOperator: StringOperators.NOOP,
+	});
+
+	const updateOrRestoreSelectColumns = (
+		key: string,
+		allViewsData: ViewProps[] | undefined,
+		options: OptionsQuery,
+		handleOptionsChange: (newQueryData: OptionsQuery) => void,
+	): void => {
+		const currentViewDetails = getViewDetailsUsingViewKey(key, allViewsData);
```
[the remark this answers is not in the document]
Comment (ahmadshaheer) [reply frontend/src/container/ExplorerOptions/ExplorerOptions.tsx:271]: Yes, @YounixM, the if is going to handle the falsy state properly. it's return type is https://github.com/SigNoz/signoz/blob/cfeb4d82d67a2843b165ba537a61a3936fb6308d/frontend/src/components/ExplorerCard/types.ts#L25-L33.

File: frontend/src/container/ExplorerOptions/ExplorerOptions.tsx [hunk 2]
```diff
@@ -252,6 +256,38 @@ function ExplorerOptions({
...
+		if (!currentViewDetails) {
+			return;
+		}
+
+		const extraData = JSON.parse(currentViewDetails?.extraData ?? '{}');
```
[the remark this answers is not in the document]
Comment (ahmadshaheer) [reply frontend/src/container/ExplorerOptions/ExplorerOptions.tsx:276]: Done.

File: frontend/src/container/ExplorerOptions/ExplorerOptions.tsx [hunk 3]
```diff
@@ -252,6 +256,38 @@ function ExplorerOptions({
...
+
+		if (!!extraData && extraData?.selectColumns) {
```
[the remark this answers is not in the document]
Comment (ahmadshaheer) [reply frontend/src/container/ExplorerOptions/ExplorerOptions.tsx:278]: Done

File: frontend/src/container/ExplorerOptions/ExplorerOptions.tsx [hunk 4]
```diff
@@ -321,6 +357,13 @@ function ExplorerOptions({
 
 		updatePreservedViewInLocalStorage(option);
 
+		updateOrRestoreSelectColumns(
+			option.key,
+			viewsData?.data?.data,
```
[the remark this answers is not in the document]
Comment (ahmadshaheer) [reply frontend/src/container/ExplorerOptions/ExplorerOptions.tsx:362]: Yes, @YounixM, the second param can be undefined.

https://github.com/SigNoz/signoz/blob/cfeb4d82d67a2843b165ba537a61a3936fb6308d/frontend/src/container/ExplorerOptions/ExplorerOptions.tsx#L270-L275

**Review by YounixM — APPROVED** [review]

**Review by SagarRajput-7 — APPROVED** [review]

**Review by YounixM — APPROVED** [review]

**Pull request merged**